networking/eth2/src/main/java/tech/pegasys/artemis/networking/eth2/rpc/core/Eth2OutgoingRequestHandler.java

```diff
@@ -135,6 +140,9 @@ private void completeRequest(final RpcStream rpcStream) {
   }
 
   private void cancelRequest(final RpcStream rpcStream, Throwable error) {
+    if (!isClosed.compareAndSet(false, true)) {
```

Comment:
Why use compare and set here instead of just setting is closed to true?

Reply:
This ensures that we only run the closing code once.  If `isClosed` is already `true`, we'll exit early.